scripts/src/blockchain/channelsPool.ts

```diff
@@ -0,0 +1,59 @@
+import {KeyPair} from "./keyPair";
+import {ChannelBusyError} from "../errors";
+
+export class ChannelsPool {
+
+	private readonly channels: Channel[] = [];
+
+	constructor(seeds: string[]) {
+		this.channels = seeds.map(seed => ({keyPair: KeyPair.fromSeed(seed), state: 'free' as ChannelState}));
+	}
+
+	private getFreeChannels(): Channel[] {
+		return this.channels.filter(channel => channel.state == 'free');
+	}
+
+	public async acquireChannel(func: (channel: Channel) => Promise<void>): Promise<void> {
+		let freeChannels = this.getFreeChannels();
+		if (freeChannels.length == 0) {
+			throw new ChannelBusyError();
+		}
+		const randomIndex = Math.floor(Math.random() * freeChannels.length);
```

Comment:
Why are you using a random index?

Reply:
Lesson learned from Ron, in order to prevent from specific channels to run out from fees while the rest still has, we try to spread over the use of channels.